src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorSpan.cs

```diff
@@ -85,8 +85,6 @@ public TensorSpan(T[]? array, int start, scoped ReadOnlySpan<nint> lengths, scop
                 this = default;
                 return; // returns default
             }
-            if (!typeof(T).IsValueType && array.GetType() != typeof(T[]))
-                ThrowHelper.ThrowArrayTypeMismatchException();
```

Comment:
Why is this being removed?

Reply:
I can add it back in, but I couldn't see the case where it would actually get hit so I removed it.

Reply:
Ok, discussed with @tannergooding offline and realize the reason for it. Added it back in and added a test for it.